### ISSUE TITLE
GPU SSD nms op cause HalideIR assert fail

### ISSUE THREAD

**Issue by janboeye** [dmlc/tvm#1532]
hi, 

in cuda/nms.py line 117, sort_ir will xor int32 with bool type, this will reject by Halide IR and cause a assert fail.
```
                    with ib.if_scope(((data_new[i+start] < data_new[i+start+1]) ^
                                      is_descend) == False):
```
I do not understand why xor less-than result with bool is denied by IR @tqchen @Laurawly 

````
/HalideIR/src/ir/IR.cpp:28: Check failed: t.bits() == 8 || t.bits() == 16 || t.bits() == 32 || t.bits() == 64 IntImm must be 8, 16, 32, or 64-bit
````
Thanks

**Comment by xqdan** [comment]
you may use int to express is_descend, instead of bool type

**Comment by tqchen** [comment]
@Laurawly can you followup on this, explicit type conversion using astype might be needed

**Comment by Laurawly** [comment]
I also encountered this bug when updated HalideIR, and I ended up not using xnor op here. I'll do a quick fix for this.

**Comment by tqchen** [comment]
@Laurawly can you update on this?

**Comment by Laurawly** [comment]
I'll send a PR which fixes this right away.

**Comment by tqchen** [comment]
Fixed by #1578
